### PR TITLE
Appointment Booking UI

### PR DESCRIPTION
**Pull request by quanphan2906**
In this PR, I have:

1. Built the booking UI
2. Built several common components, configure common CSS style, and more

## Booking UI
The booking UI is styled and works according to the [Figma design](https://www.figma.com/file/mXr8cXGpoN6aN7UEwApenm/Social-Economy-Action-Research-Foundation?type=design&node-id=2605-4448&mode=design&t=Pt7tXILRXaOPtvQ1-0). 

The flow starts from when the user lands on the calendar of a specific consultant they have chosen (any steps before this should be handled by the Consultant Profile team, cc @MikeProduce and @drkchn ) and ends with the booking confirmation page. 

One point to note: For now, from the payment page, the user will go directly to the booking review page, but according to the Figma design, there user flow here is supposed to branch into 3 payment flows before re-uniting at the booking review page. This should be handled by the Paymen team, cc @benkim2284 and @rayyanahmed021.

All the routes for the booking UI is put into `app/(booking)/`. All the components used for the UI is put into `components/booking`.

## Common component, common style, and more
- I have built several common components like Button, Modal, Loading, and put them into `components/common`. If you intend to add more common components, please add them here. Feel free to modify any common components as you see fit, but please inform everyone about this in your PR.
 
- The common style is put into `styles`. Whenever you are going to style a component, please import and use values from `styles/theme` instead of writing magic values. (For example, instead of writing `padding: 16`, please import `theme` and write `padding: theme.spacing.medium` instead). This is for consistency. Any modifications to the style should be discussed with the team.

- All the React Context should be put into `context/` folder.

- All helper functions should be put into `utils/` folder

### CODE REVIEW

**Comment by MikeProduce** [comment]
I opened VSCode and took a thorough look at the code. I must say, great job on setting up the styles and a structure we can all follow It will undoubtedly make various tasks much more manageable.

However, I did come across a few linting errors. In the file
 SEMS\app(booking)\payment.js at line 28:8
,there's a warning stating that 'service' is assigned a value but never used. It seems like it might be intended for future use.

Additionally, in 
SEMS\components\booking\review\ConfirmBlock.js at line 19:2
, there's an error indicating a duplicate key 'detailRow'. Please double-check this part as duplicate keys can cause unexpected behavior.

Lastly, in 
SEMS\components\common\Header.js at line 2:52
, there's a warning that 'Image' is defined but never used. It might be worth confirming whether this import is necessary or can be safely removed.

Apart from these linting errors, I had no issues understanding the code. Kudos! I plan to have at least one more person review it, but overall, it looks good to me. Thanks, Quan!

**Comment by quanphan2906** [comment]
@MikeProduce Hey Miguel. Thanks for the suggestions. I have just made a new commit to reflect the changes you suggest. Feel free to scrutinize the code for flaws in logic, functionality, or the overall structure of different components!

If you think the PR is good, you can approve it! It will allow me to merge the PR into `dev`, but I will also wait for another approval before merging, since this is so big a change

**Comment by MikeProduce** [comment]
sounds good ill merge !

**Review by MikeProduce — APPROVED** [review]

File: app/(booking)/BookingQuestions.js
Comment (euniceyoh) [inline app/(booking)/BookingQuestions.js]: i think you need to rename the `payment.js` page to `Payment.js` for this route to work

**Review by euniceyoh — COMMENTED** [review]
i think the pages & way you've organized looks good. one thing i noticed when trying this out on expo go (on iOS) is that the tab bar seems to sometimes cover the page (for ex: on booking confirmation, you can scroll down to see "cancel meeting" but it doesn't persist). it seems like a general layout issue though, so we could fix that in a separate PR. i can also look into how to fix that. otherwise, changes look good to me.

**Comment by quanphan2906** [comment]
@euniceyoh Just made a commit to reflect your comments. If you're deem the changes are good, give me an approval ^^

About the fact that the Tab bar blocked the bottom of the page, I added a quick fix by adding another layer of `ScrollView` as a wrapper for each component in booking. Not sure if it's the most sustainable one, but it does the trick for now. We can easily revert it in the future. It's a small issue so we shouldn't spend too much time on it, I would suggest.

**Review by euniceyoh — APPROVED** [review]

**Pull request merged**
